### PR TITLE
UI: Add connection status square to status bar

### PR DESCRIPTION
**Pull request by cg2121**
This adds the connection status square. This was in OBS Classic.

### CODE REVIEW

**Comment by ONSBalder** [comment]
Can one of the admins verify this patch?

**Comment by notr1ch** [comment]
I would prefer if this only looked at connection metrics as Classic does. For example, skipped frames should not affect the status square, as this is a CPU usage problem not a connectivity problem.

**Comment by cg2121** [comment]
In OBS Classic the square would also show up if you were only recording, not just streaming. That's why I included the skipped frames metrics.

**Comment by cg2121** [comment]
I made it now that it only looks at the connection.

File: UI/window-basic-status-bar.cpp [hunk 1]
```diff
@@ -127,15 +164,49 @@ void OBSBasicStatusBar::UpdateBandwidth()
...
+
+	if (noBitrate) {
+		QPixmap pixmap(16, 16);
+		pixmap.fill(QColor("transparent"));
+		QPainter painter(&pixmap);
+		painter.setPen(Qt::NoPen);
+		painter.setBrush(QBrush(QColor(255,0,0)));
+		painter.drawRect(0, 0, 16, 16);
+		statusSquare->setPixmap(pixmap);
+	}
```
Comment (RytoEX) [inline UI/window-basic-status-bar.cpp]: Style:  Curly braces should be on the same line as non-function statement blocks (if, else if, else, switch, for, while, do).  See Line 191-192 and 200-201.

**Pull request closed without merging**